src/eureka/S3_data_reduction/niriss.py

```diff
@@ -1,698 +1,503 @@
-
-        if cutends is not None:
-            centers[cutends:] = 0
-
-        return centers
-
-    def clean_and_fit(x1, x2, y1, y2):
-        x1, y1 = x1[y1 > 0], y1[y1 > 0]
-        x2, y2 = x2[y2 > 0], y2[y2 > 0]
-
-        poly = np.polyfit(np.append(x1, x2),
-                          np.append(y1, y2),
-                          deg=4)  # hard coded deg of polynomial fit
-        fit = np.poly1d(poly)
-        return fit
-
-    g = simplify_niriss_img(data, meta, isplots)
-
-    f, _ = f277_mask(data)
-
-    # g_centers = find_centers(g, cutends=None)
-    f_centers = find_centers(f, cutends=430)  # hard coded end of the F277 img
-
-    gcenters_1 = np.zeros(len(g[0]), dtype=int)
-    gcenters_2 = np.zeros(len(g[0]), dtype=int)
-
-    for i in range(len(g[0])):
-        inds = np.where(g[:, i] > 100)[0]
-        inds_1 = inds[inds <= 78]  # hard coded y-boundary for the first order
-        inds_2 = inds[inds >= 80]  # hard coded y-boundary for the second order
-        if len(inds_1) >= 1:
-            gcenters_1[i] = np.nanmean(inds_1)
-        if len(inds_2) >= 1:
-            gcenters_2[i] = np.nanmean(inds_2)
-
-    gcenters_1 = rm_outliers(gcenters_1)
-    gcenters_2 = rm_outliers(gcenters_2)
-    x = np.arange(0, len(gcenters_1), 1)
-
-    fit1 = clean_and_fit(x, x[x > 800],
-                         f_centers, gcenters_1[x > 800])
-    fit2 = clean_and_fit(x, x[(x > 800) & (x < 1800)],
-                         f_centers, gcenters_2[(x > 800) & (x < 1800)])
-
-    if isplots >= 5:
-        plt.figure(figsize=(14, 4))
-        plt.title('Order Approximation')
-        plt.imshow(g+f)
-        plt.plot(x, fit1(x), 'k', label='First Order')
-        plt.plot(x, fit2(x), 'r', label='Second Order')
-        plt.xlabel('x')
-        plt.ylabel('y')
-        plt.legend(bbox_to_anchor=(1.05, 1), loc=2, borderaxespad=0.)
-        plt.show()
-
-    tab = Table()
-    tab['x'] = x
-    tab['order_1'] = fit1(x)
-    tab['order_2'] = fit2(x)
-
-    if save:
-        tab.write('niriss_order_fits_method1.csv', format='csv')
-
-    meta.tab1 = tab
-
-    return meta
-
-
-def mask_method_two(data, meta, isplots=0, save=False):
-    """
-    A second method to extract the masks for the first and
-    second orders in NIRISS data. This method uses the vertical
-    profile of a summed image to identify the borders of each
-    order.
+    '''
+    # Mask trace regions from other orders
+    for order in meta.orders:
+        trace = np.round(data.trace.sel(order=order).values).astype(int)
+        wave = data.wave_1d.sel(order=order).values
+        other_orders = list.copy(meta.orders)
+        other_orders.remove(order)
+        for j in np.where(~np.isnan(wave))[0]:
+            ymin = np.max((0, trace[j] - meta.spec_hw))
+            ymax = np.min((trace[j] + meta.spec_hw + 1, len(data.y) + 1))
+            for other_order in other_orders:
+                data['mask'].sel(order=other_order)[:, ymin:ymax, j] = True
```

Comment:
This feels like it should be in a separate function since it isn't really related to straightening traces.